### ISSUE TITLE
Error when predicting on Big Data

### ISSUE THREAD

**Issue by rominasilvera** [USEPA/spmodel#13]
Hi.
I was running a model, the training dataset (sf object) contains 62.770 observations. I used:

`
mod_0 <- splm( formula_general, data = training(split), spcov_type = "exponential", partition_factor = ~ clean_uso, local = pred_list )
`

where `pred.list` is:
`
pred_list <- list( parallel = TRUE, ncores = 4 )
`

Training the model took around 15 minutes.
When using `predict()` (or `augment()`)

The code throws the following error:

`pred.mod_0 <- augment( mod_0,  newdata = testing(split), local = TRUE ) 
Error in x$terms %||% attr(x, "terms") %||% stop("no terms component nor attribute") :  no terms component nor attribute
`

Or when setting `local` to `pred_list`

`
pred.mod_0 <- augment(
mod_0,
newdata = testing(split),
local = pred_list
)
Error in checkForRemoteErrors(val) :
4 nodes produced errors; first error: no terms component nor attribute
`

Could you please provide help?

Here's more info about the objects I use:

`
> class(dataFormat)
[1] "sf"         "data.frame"
> dim(dataFormat)
[1] 89673    30

split <- initial_split(
  dataFormat,
  prop = 0.7, 
  strata = "clean_uso")

> print(formula_general)
y_valor ~ x_renta + x_val_m + x_super + x_catsu + garaje + terraza + 
    dormitorios + aseos
`

There are no missing values

`
sessionInfo() R version 4.3.2 (2023-10-31 ucrt) Platform: x86_64-w64-mingw32/x64 (64-bit) Running under: Windows 11 x64 (build 22621) Matrix products: default locale: [1] LC_COLLATE=Spanish_Spain.utf8 LC_CTYPE=Spanish_Spain.utf8 LC_MONETARY=Spanish_Spain.utf8 [4] LC_NUMERIC=C LC_TIME=Spanish_Spain.utf8 time zone: Europe/Madrid tzcode source: internal attached base packages: [1] stats graphics grDevices utils datasets methods base other attached packages: [1] spmodel_0.5.0 latex2exp_0.9.6 ggpubr_0.6.0 visdat_0.6.0 gstat_2.1-1 pdp_0.8.1 [7] vip_0.4.1 Metrics_0.1.4 recipes_1.0.8 forecast_8.21.1 rsample_1.2.0 caret_6.0-94 [13] lattice_0.21-9 ggplot2_3.4.4 jsonlite_1.8.7 mongolite_2.7.2 leaflet_2.2.0 sf_1.0-14 [19] skimr_2.1.5 corrplot_0.92 ggthemes_4.2.4 kableExtra_1.3.4 formattable_0.2.1 data.table_1.14.8 [25] tidyr_1.3.0 dplyr_1.1.3 stringr_1.5.0 httr_1.4.7 sp_2.1-1 loaded via a namespace (and not attached): [1] RColorBrewer_1.1-3 wk_0.9.0 rstudioapi_0.15.0 magrittr_2.0.3 farver_2.1.1 [6] rmarkdown_2.25 vctrs_0.6.4 askpass_1.2.0 base64enc_0.1-3 rstatix_0.7.2 [11] webshot_0.5.5 htmltools_0.5.7 curl_5.1.0 broom_1.0.5 s2_1.1.4 [16] xgboost_1.7.5.1 pROC_1.18.5 TTR_0.24.3 parallelly_1.36.0 KernSmooth_2.23-22 [21] htmlwidgets_1.6.2 plyr_1.8.9 zoo_1.8-12 lubridate_1.9.3 lifecycle_1.0.4 [26] iterators_1.0.14 pkgconfig_2.0.3 Matrix_1.6-1.1 R6_2.5.1 fastmap_1.1.1 [31] future_1.33.0 digest_0.6.33 colorspace_2.1-0 furrr_0.3.1 warp_0.2.1 [36] crosstalk_1.2.0 labeling_0.4.3 fansi_1.0.5 timechange_0.2.0 abind_1.4-5 [41] compiler_4.3.2 intervals_0.15.4 proxy_0.4-27 bit64_4.0.5 withr_2.5.2 [46] backports_1.4.1 tseries_0.10-54 carData_3.0-5 DBI_1.1.3 ggsignif_0.6.4 [51] MASS_7.3-60 lava_1.7.3 openssl_2.1.1 classInt_0.4-10 ModelMetrics_1.2.2.2 [56] tools_4.3.2 units_0.8-4 lmtest_0.9-40 quantmod_0.4.25 future.apply_1.11.0 [61] nnet_7.3-19 glue_1.6.2 quadprog_1.5-8 nlme_3.1-163 grid_4.3.2 [66] stringdist_0.9.10 mapSpain_0.8.0 reshape2_1.4.4 generics_0.1.3 gtable_0.3.4 [71] class_7.3-22 RPostgres_1.4.6 hms_1.1.3 car_3.1-2 xml2_1.3.5 [76] utf8_1.2.4 ggrepel_0.9.4 foreach_1.5.2 pillar_1.9.0 splines_4.3.2 [81] FNN_1.1.3.2 survival_3.5-7 bit_4.0.5 tidyselect_1.2.0 knitr_1.45 [86] gridExtra_2.3 urca_1.3-3 svglite_2.1.2 stats4_4.3.2 xfun_0.41 [91] hardhat_1.3.0 timeDate_4022.108 stringi_1.7.12 evaluate_0.23 codetools_0.2-19 [96] tibble_3.2.1 cli_3.6.1 rpart_4.1.21 systemfonts_1.0.5 jquerylib_0.1.4 [101] repr_1.1.6 munsell_0.5.0 spacetime_1.3-0 Rcpp_1.0.11 globals_0.16.2 [106] slider_0.3.1 parallel_4.3.2 ellipsis_0.3.2 fracdiff_1.5-2 gower_1.0.1 [111] blob_1.2.4 listenv_0.9.0 viridisLite_0.4.2 ipred_0.9-14 scales_1.2.1 [116] xts_0.13.1 prodlim_2023.08.28 e1071_1.7-13 purrr_1.0.2 rlang_1.1.2 [121] rvest_1.0.3
`

**Comment by michaeldumelle** [comment]
Thanks @rominasilvera . The bug occurred when performing local (i.e., big data) prediction using a partition factor. This has been fixed in the development version of `spmodel`, which you can download by running

```r
remotes::install_github("USEPA/spmodel", ref = "develop")
```

This fix will be reflected in the next CRAN submission, which should be sometime within the next few months (the current version of `spmodel` is 0.5.0).

**Comment by michaeldumelle** [comment]
@rominasilvera the fix is now on CRAN.
